Assets/Scripts/Tools/EdgeTool.cs

```diff
@@ -34,11 +34,49 @@ public void OnLeftClick(Transform cameraTransform, bool isHit, RaycastHit raycas
       if (!isHit || raycastHit.collider.gameObject.tag != Constants.PhysicalNodeTag)
         return;
 
-      OnActionPerformed(isHit, raycastHit);
+      if (!isHit)
+        return;
+
+      var gameObjectHit = raycastHit.collider.gameObject;
+      var currentlyHitNode = _graphService.FindNodeByGameObject(gameObjectHit);
+
+      // If not a node, don't do anything
+      if (currentlyHitNode == null)
+        return;
+
+      if (_previouslyHitNode == null)
+      {
+        _previouslyHitNode = currentlyHitNode;
+        EnablePreviouslyHitNode();
+        return;
+      }
+
+      _graphService.AddEdge(currentlyHitNode, _previouslyHitNode);
+      DisablePreviouslyHitNode();
+    }
+
+    private void EnablePreviouslyHitNode()
```

Comment:
I don't like the words enable/disable here, those methods only change the glow. Consider changing

Reply:
any ideas for the name?

Reply:
I think EnableSelectedNodeGlow would work